environments/hca/terraform/pubsub.tf

```diff
@@ -0,0 +1,19 @@
+module ebi_staging_notification_pubsub_topic {
+  source     = "terraform-google-modules/pubsub/google"
+  version    = "~>1.8"
+  project_id = local.dev_project_name
+  topic      = "broad-dsp-monster-hca-dev.staging-transfer-notifications.ebi"
+  pull_subscriptions = [
+    {
+      name = "ebi-writer"
+    }
+  ]
+}
+
+# EBI can consume from the EBI transfer notifications pull subscription
+resource google_pubsub_subscription_iam_member ebi_writer_iam {
```

Comment:
@raaidbroad @smark88 I want the ebi writer SA to have `roles/pubsub.subscriber` attached for the configured subscription, I believe this should get us there but I'd appreciate your 👀

Reply:
@aherbst-broad I think we also need to grant roles/pubsub.publisher for the transfer service bit to work as desired, as seen here: https://cloud.google.com/storage-transfer/docs/pub-sub-transfer?hl=en

Reply:
@raaidbroad take a look at the plan, there is a pubsub publisher service account receiving the desired role:

```
  # module.ebi_staging_notification_pubsub_topic.google_pubsub_topic_iam_member.pull_topic_binding[0] will be created
  + resource "google_pubsub_topic_iam_member" "pull_topic_binding" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = "serviceAccount:service-104615930137@gcp-sa-pubsub.iam.gserviceaccount.com"
      + project = "broad-dsp-monster-hca-dev"
      + role    = "roles/pubsub.publisher"
      + topic   = "projects/broad-dsp-monster-hca-dev/topics/broad-dsp-monster-hca-dev.staging-transfer-notifications.ebi"
    }
```

Judging by the docs I believe this should suffice, as the ebi-writer acct is presumably not actually the principal performing the publish to the topic (but it's not 100% clear from their docs tbh).

Reply:
Ahhh got it, awesome.